### ISSUE TITLE
Deathadder V2 profile switch does switch illumination

### ISSUE THREAD

**Issue by LupeChristoph** [openrazer/openrazer#1630]
There are actually two issues with the profile switch. The first is that on windows 10, it switches between the initialized profiles, in my case red and green (the colors are white, red, green, blue and cyan). In openrazer, it includes the "white" profile. But that is just the running copy of the profile selected last. Please omit that.

Now, when the profiles are switched, the button function of that selected profile are activated (e.g. hypershift on button 5 and hypershift changes the scroll wheel to vol+, vol-, and mute. But the colors of the wheel and the logo are still cycling as the default does.

Same goes for macros and text on buttons, but there Razer Synapse warns that they require Synapse to work. So it is quite possible that color control also requires it, but doesn't tell.

\<rant>It seems that the mouse is at the mercy of the software. I despise this solution. It is quite possible to design a mouse that is fully functional without software running on the host. I might send the Deathadder back and get a new Roccat Kone. The plastic on my old Kone is deteriorating and I need a new mouse. And I require Hypershift or Easy Shift or whatever Logitech calls it, which limits the options.\</rant>

**Comment by z3ntu** [comment]
> There are actually two issues with the profile switch.

What's "the profile switch"?

> In openrazer, it includes the "white" profile. But that is just the running copy of the profile selected last. Please omit that.

Please elaborate on that. We don't have any "profiles" in openrazer.

> So it is quite possible that color control also requires it, but doesn't tell.

Sounds like it.

> <rant>It seems that the mouse is at the mercy of the software.

Correct, if you don't like it, don't buy Razer products, nothing I can do about it. :shrug:

**Comment by LupeChristoph** [comment]
> > There are actually two issues with the profile switch.
> What's "the profile switch"?

The Deathadder V2 has a switch on the bottom side that switches the profiles. And a color LED that indicates which profile is active. That LED cycles red, green, blue and cyan when the mouse is connected to Windows with Synapse. It cycles white, red, green, blue and cyan when the mouse is connected to Ubuntu. So there is a differenec, which may just be the absence of Synapse.

> > In openrazer, it includes the "white" profile. But that is just the running copy of the profile selected last. Please omit that.
> Please elaborate on that. We don't have any "profiles" in openrazer.

The profiles contain fours different sets of mouse settings, like button mapping, illumination and macros. Part of this (as far as I'm aware only the button mappings to simple key strokes or mouse button clicks) is stored in the mouse ans is usable independent of Synapse. The rest is handled by Synapse, so it is not available unless Synapse is running and able to contact the mouse. I have not tried, but I'm certain even a normal KVM switch is sufficient to disable this when the mouse is routed through it.
 
> > It seems that the mouse is at the mercy of the software.
> Correct, if you don't like it, don't buy Razer products, nothing I can do about it. shrug

Sure, I'm aware of that, and the mouse is on its way back to Amazon. It got a newer Roccat Kone mouse, and that demonstrates that all mouse feature, including illumination and macros can be handled by the mouse with assistance of host-based software. The funny thing is that Roccat decided to abandon that with their current keyboards. I talked to their support before I ordered it, and the support told me that all the the keyboard features will work without host software. D'uh! Keyboard goes back. I've got a Steelseries keyboard now, and that demonstrates the same thing, that host-based software can be unnecessary unless you want to manage the settings of the device.
